cpp/src/arrow/compute/kernels/scalar_arithmetic_benchmark.cc

```diff
@@ -62,16 +62,23 @@ static Result<Datum> DivideChecked(const Datum& left, const Datum& right,
   return Divide(left, right, std::move(options), ctx);
 }
 
+static Result<Datum> PowerChecked(const Datum& left, const Datum& right,
+                                  ArithmeticOptions options = ArithmeticOptions(),
+                                  ExecContext* ctx = NULLPTR) {
+  options.check_overflow = true;
+  return Power(left, right, std::move(options), ctx);
+}
+
 template <BinaryOp& Op, typename ArrowType, typename CType = typename ArrowType::c_type>
 static void ArrayScalarKernel(benchmark::State& state) {
   RegressionArgs args(state);
 
   const int64_t array_size = args.size / sizeof(CType);
 
   // Choose values so as to avoid overflow on all ops and types
-  auto min = static_cast<CType>(6);
-  auto max = static_cast<CType>(min + 15);
-  Datum rhs(static_cast<CType>(6));
+  auto min = static_cast<CType>(1);
+  auto max = static_cast<CType>(min + 4);
+  Datum rhs(static_cast<CType>(3));
```

Comment:
I had to reduce the range to avoid overflows. Would it make more sense to introduce a separate generator for power benchmark only?

Reply:
I tweaked some code and removed the benchmark.
For now, I think there are still many things needs to improve. So we don't need to add benchmark in a haste.
I downloaded and attached your benchmark commit, in case it's lost.
[0001-Adding-benchmark-for-Power-and-PowerChecked.patch.gz](https://github.com/apache/arrow/files/6301530/0001-Adding-benchmark-for-Power-and-PowerChecked.patch.gz)